src/controller/DevicePairingDelegate.h

```diff
@@ -70,6 +70,12 @@ class DLL_EXPORT DevicePairingDelegate
      *   Called when the commissioning process is complete (with success or error)
      */
     virtual void OnCommissioningComplete(NodeId deviceId, CHIP_ERROR error) {}
+
+#if CHIP_DEVICE_CONFIG_ENABLE_JOINT_FABRIC
+    virtual void OnCommissioningComplete(NodeId deviceId, Optional<Crypto::P256PublicKey> trustedIcacPublicKeyB, CHIP_ERROR error)
```

Comment:
Why the "B" at the end of the name?

And more importantly, where is the documentation.  When is this called?  Is it called in addition to the other OnCommissioningComplete?  Or instead of?  Or something else?

What I would expect is that there is only one OnCommissioningComplete call, with this one calling the other if not overridden or something.

Finally, having vtables that depends on the comple options like this is really fragile and fails catastrophically if somehow things disagree on the vtable layout.  it should really be avoided.

Reply:
_trustedIcacPublicKeyB_ is also the terminology used in the spec (see [here](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/multi_admin/multi_admin.adoc?plain=1#L176)) so I am reusing it here. It represents the public key bytes of the ICAC of the commissionee Admin.

Changed the logic such that there is a single call to `OnCommissioningComplete` inside CHIPDeviceController.cpp and added documentation for this new call. Agree with the comment around vtables.